### ISSUE TITLE
Allow `imp Trait` in trait method

### ISSUE THREAD

**Issue by xmh0511** [rust-lang/rust#105440]
````rust
trait MyTrait{
   fn show(&self)-> impl Trait
}
````
In the current, `impl Trait` as the return type is not allowed in trait method, however, it's useful and should be consistent with other functions/methods(i.e permit `impl Trait` in the return type). https://internals.rust-lang.org/t/impl-trait-should-be-supported-in-the-return-type-of-a-closure/17917/6

**Comment by compiler-errors** [comment]
Gonna close this since it's not really an issue that can be addressed (since it's just an implemented feature that's waiting on discussion and stabilization) -- if you find an issue with `#![feature(return_position_impl_trait_in_trait)]`, feel free to open another issue!
